src/main/java/org/javarosa/xpath/expr/XPathEqExpr.java

```diff
@@ -50,15 +50,21 @@ public void readExternal(DataInputStream in, PrototypeFactory pf) throws IOExcep
             op = XPathEqExpr.NEQ;
         }
         computedCacheability = ExtUtil.readBool(in);
-        isCacheable = ExtUtil.readBool(in);
+        exprIsCacheable = ExtUtil.readBool(in);
```

Comment:
It feels like there's a lot of context  and state that's being managed here that  would be much cleaner to delegate to a composed cacheState object (which would also make it easier to change this later more safely without requiring safely updating 5-6 implementaitons

Reply:
yea that's a really good point, sorry I should have thought of that